### ISSUE TITLE
Highlight relevant passages in document

### ISSUE THREAD

**Issue by RealLitb** [stanford-futuredata/ColBERT#394]
Please bear with me, I'm not too experienced with machine learning and ColBERT.

I wanted to reproduce https://colbert.aiserv.cloud/ and use this ColBERT library with the colbert-ir/colbertv2.0 model. Therefore I looked at various resources on the internet and came up with this code, partially with help from ChatGPT:

```
from colbert.infra import Run, RunConfig, ColBERTConfig
from colbert.modeling.checkpoint import Checkpoint
import torch

# Step 1: Set up ColBERT configuration and tokenizer
checkpoint_path = "colbert-ir/colbertv2.0"  # Pre-trained checkpoint
config = ColBERTConfig(
  checkpoint=checkpoint_path
)

# Step 2: Load the ColBERT model
model = Checkpoint(checkpoint_path, colbert_config=config)
model.eval()

# Step 3: Define query and passage
query = "What is the capital of France?"
passage = "France is a country in Europe, and its capital is Paris."

# Step 5: Encode query and passage
query_embs = model.queryFromText([query])
passage_embs = model.docFromText([passage])

# Step 6: Compute token-wise similarity
similarity_matrix = passage_embs @ query_embs.transpose(1, 2)  # [1, passage_len, query_len]
print("similarity_matrix:\n", similarity_matrix)

relevance_scores = similarity_matrix.max(dim=2).values.squeeze(0)  # [passage_len]
print("Relevance scores:\n", relevance_scores.tolist())
```

Expected to get a similar scoring like the one shown in "colbert.aiserv.cloud" for the Query and Passage, but to my surprise, the token "Paris" get a very low score ("0.585745632648468"), which is even lower than "is". 

What am I doing wrong? Should I weight the tokens using some kind of IDF afterwards to lower the score of common words? Does the referenced page do something like that?

**Comment by okhat** [comment]
You're probably forgetting about the padding tokens like [Q] and [D]?
